main.go

```diff
@@ -537,19 +537,39 @@ func main() {
 	}
 
 	// Simulator infos
-	simulator, err := xcodeutil.GetSimulator(configs.SimulatorPlatform, configs.SimulatorDevice, configs.SimulatorOsVersion)
-	if err != nil {
+	var sim simulator.InfoModel
+	var osVersion string
+	var errGetSimulator error
+	if configs.SimulatorOsVersion == "latest" {
+		simulatorPlatformSplit := strings.Split(configs.SimulatorPlatform, " Simulator")
+		if len(simulatorPlatformSplit) == 0 {
+			errGetSimulator = fmt.Errorf("failed to parse simulator platform (%s)", configs.SimulatorPlatform)
+		}
+
+		var simulatorDevice = configs.SimulatorDevice
+		if simulatorDevice == "iPad" {
+			log.Warnf("Given device (%s) is deprecated, using (iPad 2)...", simulatorDevice)
+			simulatorDevice = "iPad 2"
+		}
+
+		desiredPlatform := simulatorPlatformSplit[0]
```

Comment:
why 0?

Reply:
That's just the index where the relevant part of the split string lives.

I moved the splitting closer to make the logic clearer.